watcher/watcher.go

```diff
@@ -61,7 +60,7 @@ func (w *Watcher) Watch() (<-chan *Event, error) {
 			}
 
 			for _, watchEvent := range resp.Events {
-				event := &Event{ID: extractIDFromKey(watchEvent.Kv.Key)}
+				event := &Event{ID: connection.ID(string(watchEvent.Kv.Key))}
```

Comment:
You shouldn't need the double-cast here, try converting straight to `connection.ID`.

Reply:
That would be too much magic.